server/rpc/TransactionRPC.java

```diff
@@ -185,25 +186,35 @@ void closeWithError(Throwable error) {
         private final ConcurrentMap<String, BatchingIterator<?>> iterators = new ConcurrentHashMap<>();
 
         /**
-         * Spin up an iterator and begin batch iterating.
+         * Spin up a new iterator and begin streaming responses immediately.
          */
-        <T> void beginIteration(TransactionProto.Transaction.Req request, Iterator<T> iterator, Function<List<T>, TransactionProto.Transaction.Res> responseBuilderFn) {
-            beginIteration(request, iterator, transaction.options().batchSize(), responseBuilderFn);
+        <T> void createNewIterator(TransactionProto.Transaction.Req request, Iterator<T> iterator, Function<List<T>, TransactionProto.Transaction.Res> responseBuilderFn) {
```

Comment:
`createNewIterator()` implies the construction of a new object and thus returning it. This method does quite the opposite.

Reply:
+1. In fact I think the purpose of these Iterators has shifted quite considerably from when they were first created. I've raised an issue here: https://github.com/graknlabs/grakn/issues/6075.

I can't think of a better name really. `beginIteration` was misleading because the 2nd overload _doesn't_ begin iteration if it's an `insert` query.

Reply:
`iterate(...)` would be an appropriate enough name here.

Reply:
Updated